### PR TITLE
fix(number-input): use label for aria-label, deprecate ariaLabel prop

### LINKED ISSUE #4499: AVT 1 - NumberInput DAP violation that accessible name does not match the visible label.
## Environment
macOS 10.14.6
Chrome Version 77.0.3865.90
VoiceOver
DAP 1.8.0.0 Sept 2019 WCAG 2.1 Ruleset

## Description
Run DAP on the React Number Input Component. The following violation is found. (see screenshot):
Screen Shot 2019-10-30 at 2 33 47 PM

Note: The issue is that the label is "Number Input label", but the aria-label="Number input field with increment and decrement buttons". If the aria-label is updated to match the actual label the violation will no longer be present.

### PR DESCRIPTION
**Pull request by jendowns**
Closes #4499

My understanding is that the label text should match the `aria-label` attribute text on the input.

But in this case, both the label and the `aria-label` are populated by separate props, so they can differ if two different strings are provided.

To resolve this, I propose deprecating the `ariaLabel` prop that populates the `aria-label` attribute. And then, I propose making it so the `label` prop also populates the `aria-label` attribute. That way, both the label and `aria-label` will always match.

### Changelog

**Changed**

- use `label` prop to populate `aria-label` attribute text
- deprecate `ariaLabel` prop

**Removed**

- remove default value for `ariaLabel` since the prop is no longer used & is deprecated

#### Testing / Reviewing

View the the PR deployment for `carbon-components-react`, view the `NumberInput` component, and run the DAP tool. Make sure that you don't see a violation for mismatched label/aria-label text, as shown in the originating issue #4499.

### CODE REVIEW

**Comment by jendowns** [comment]
Tagging @abbeyhrt & @dakahn ☝️

File: packages/react/src/components/NumberInput/NumberInput.js [hunk 1]
```diff
@@ -137,7 +138,10 @@ class NumberInput extends Component {
     /**
      * Provide a description that would be used to best describe the use case of the NumberInput component
      */
-    ariaLabel: PropTypes.string,
+    ariaLabel: deprecate(
+      PropTypes.string,
+      `\nThe prop \`ariaLabel\` for NumberInput has been deprecated in favor of \`label\`.`
```
Comment (joshblack) [inline packages/react/src/components/NumberInput/NumberInput.js]:
```suggestion
      `\nThe prop \`ariaLabel\` for NumberInput has been deprecated in favor of \`aria-label\`.`
```

**Comment by joshblack** [comment]
Might have jumped the gun on my review, sorry about that. Should we use `aria-label`? Using `label` was confusing initially which is why I thought it might have been a mix-up.

**Review by joshblack — COMMENTED** [review]
Commenting to remove review status

**Comment by jendowns** [comment]
@joshblack I happened to see your original comment before the review was removed so I just wanted to address it in case I'm misunderstanding something --

Since the label (which is always there, even if you have `hideLabel={true}` it is merely visually hidden) on the input and the `aria-label` attribute on the input need to match, I don't see how there can be separate props for both. So that's why I'm thinking it's best to deprecate the `ariaLabel` prop and instead use the `label` prop to populate the input's label element AND its `aria-label` attribute. As it stands, I don't think that the existing `label` and `ariaLabel` props can coexist together. But please let me know if you disagree -- thank you!

**Comment by joshblack** [comment]
Just to make sure I have this right:

- We need an accessible label for `<input>`
- If `label` prop is provided, we have a `<label>` that associates the label with `<input>`
- If no `label` is provided, we have `ariaLabel` prop on the `<input>` node
- If both are supplied, we may end up violating [Checkpoint 2.5.3](https://www.ibm.com/able/guidelines/ci162/label_in_name_71.html)

If this is the case, would it make sense to specify that only one should be supplied, versus both?

**Comment by joshblack** [comment]
@jendowns sorry about that! Was trying to remove since it was clear I didn't have enough context into the issue. Commented above to see if I understood things correctly 😬

**Comment by jendowns** [comment]
@joshblack No worries!

About specifying that one or the other be applied — I just realized that the label prop isn’t actually required. Whoops 😂

Okay so... we should encourage one prop or the other, which would likely involve just updating the storybook demo really? And not deprecating anything?

**Comment by jendowns** [comment]
Actually, I think I may close this PR. 😅 My approach here isn't addressing the full situation. I think a couple decisions need to be made around `label` and `hideLabel` in particular --

If the purpose of `hideLabel` is to visually hide the label, but keep the text readable for screen readers, then shouldn't `label` just be required? And then `aria-label` removed completely, as it would no longer be needed?

I think `label`, `hideLabel`, and `ariaLabel` props needs to "interact" differently, otherwise this conflict will continue.

But again, my approach here doesn't resolve things completely! So I'm going to close this one out. Thanks for the notes!

**Pull request closed without merging**